src/agent/session.rs

```diff
@@ -87,7 +87,7 @@ impl Session {
 impl SessionBuilder {
     /// Starts a session
     #[tracing::instrument(skip_all)]
-    pub async fn start(&mut self, index: &impl Index) -> Result<RunningSession> {
+    pub async fn start(&mut self, index: &(impl Index + 'static)) -> Result<RunningSession> {
```

Comment:
Any lingering 'static should be remove as it's now part of the trait definition.